pos_auto_invoice/static/src/js/models.js

```diff
@@ -0,0 +1,20 @@
+odoo.define("pos_auto_invoice.models", function (require) {
+    "use strict";
+
+    const {Order} = require("point_of_sale.models");
+    const Registries = require("point_of_sale.Registries");
+
+    // eslint-disable-next-line no-shadow
+    const AutoInvoiceOrder = (Order) =>
+        // eslint-disable-next-line no-shadow
+        class AutoInvoiceOrder extends Order {
+            constructor() {
+                super(...arguments);
+                if (this.pos.config.invoice_by_default) {
+                    this.to_invoice = true;
+                }
+            }
+        };
```

Comment:
Nice to have : not blocking : consider the following use case : 
- create a new order (invoice is set to True)
- user uncheck the "to invoice"
- reload the PoS
- the order is checked again

overload export_from_JSON / load_from_JSON should be fix the problem I guess.